// Copyright 2021 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package tfx.orchestration;

import "google/protobuf/timestamp.proto";
import "ml_metadata/proto/metadata_store.proto";

// Connection config for ML Metadata.
message MLMDConnectionConfig {
  reserved 3;
  oneof connection_config {
    ml_metadata.ConnectionConfig database_connection_config = 1;
    ml_metadata.MetadataStoreClientConfig mlmd_store_client_config = 2;
  }
}

// TODO(b/256921318) Revisit if this message is the best way to save MLMD db
// info in pipeline IR for cross project producer/consumer pipelines.
message MLMDServiceConfig {
  string owner = 1;
  string name = 2;
  string mlmd_service_target = 3;
}

// Filter options for getting MLMD nodes (Artifacts / Executions / Contexts)
// from MLMD.
message NodeFilterOptions {
  // The earliest create time (inclusive) of MLMD nodes.
  // If not set, will consider all MLMD nodes created since the Unix epoch.
  google.protobuf.Timestamp min_create_time = 1;

  // The latest create time (inclusive) of MLMD nodes.
  // If not set, will consider all MLMD nodes created before now.
  google.protobuf.Timestamp max_create_time = 2;

  // The names of node types (ArtifactTypes / ExecutionTypes / ContextTypes).
  //
  // Examples are:
  // 'Model' as the name of an ArtifactType,
  // 'tfx.dsl.components.common.resolver.Resolver' as the name of an
  // ExecutionType,
  // 'pipeline' as the name of a ContextType.
  //
  // If not set, will consider all MLMD nodes regardless of their types.
  repeated string types = 3;
}
